scripts/setup_macos.sh

```diff
@@ -27,3 +27,5 @@ brew_install_if_missing libcypher-parser
 # https://github.com/KatanaGraph/homebrew-dependencies
 #brew tap KatanaGraph/dependencies
 brew uninstall apache-arrow || true
+
+pip3 install PyGithub packaging
```

Comment:
What are your thoughts on venv, poetry or pipenv? The last one we have used elsewhere to manage python dependencies.

Reply:
I would prefer to use Conda if we want a python environment tool. We are probably forced to use it anyway because it's a very common system in data science. If you just want a dependency file instead of having the package names in the script we can use a pip requirements file.

Reply:
How about a requirements file (for each python app) and some `find -name requirements.txt | xargs pip install -r ` thing at the least? This would at least allow dependencies to be written closer to the script that needs them.

Reply:
That could be useful. I would produce a single file to use to install the packages from multiple inputs. This file would probably need to be committed to the repo (like a lock file). If we are going that direction, I think just having lock files might be better: https://github.com/conda-incubator/conda-lock. It's somewhat experimental, but I played with it and it works and doesn't need to be installed on the user machine since it generates a lock file that can be used by conda directly. Installing with such a lock file also reduces the install time for one a katana-open development environment from ~60sec to ~10sec.